Microsoft.VCRTForwarders.140.targets

```diff
@@ -4,34 +4,91 @@
+      <Platform ParameterType="System.String" Required="true" />
+      <TaskOutput ParameterType="System.Boolean" Output="true" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include="System.Xml"/>
+      <Reference Include="Microsoft.Build"/>
+      <Using Namespace="Microsoft.Build" />
+      <Using Namespace="Microsoft.Build.Evaluation" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
 
-  <!-- x64 -->
-  <ItemGroup Condition="'$(Configuration)' == 'Debug' And '$(Platform)' == 'x64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-x64\native\debug\*.dll" />
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-x64\native\release\*.dll" />
-  </ItemGroup>
-   <ItemGroup Condition="'$(Configuration)' == 'Release' And '$(Platform)' == 'x64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-x64\native\release\*.dll" />
-  </ItemGroup>
+        var properties = new Dictionary<string, string>
+        {
+          { "Configuration", Configuration },
+          { "Platform", Platform }
+        };
+        
+        // Per MSDN, _DEBUG define can be checked to determine if debug CRT is in use.
+        var collection = new ProjectCollection(properties);
+        var project = collection.LoadProject(ProjectFile);
+        ProjectMetadata def = project.AllEvaluatedItemDefinitionMetadata
+                                     .LastOrDefault(p => p.ItemType == "ClCompile" && p.Name == "PreprocessorDefinitions");
+        Boolean useDebug = def != null && (";" + def.EvaluatedValue + ";").Contains(";_DEBUG;");
 
-  <!-- arm64 -->
-  <ItemGroup Condition="'$(Configuration)' == 'Debug' And '$(Platform)' == 'ARM64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\debug\*.dll" />
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\release\*.dll" />
-  </ItemGroup>
-   <ItemGroup Condition="'$(Configuration)' == 'Release' And '$(Platform)' == 'ARM64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\release\*.dll" />
-  </ItemGroup>
+        // There seem to be cases where even if _DEBUG is not found, debug CRT is used based on RuntimeLibrary.
+        if(!useDebug)
+        {
+          ProjectMetadata runtimeLibrary = project.AllEvaluatedItemDefinitionMetadata
+                                                  .LastOrDefault(p => p.ItemType == "ClCompile" && p.Name == "RuntimeLibrary");
+          useDebug = runtimeLibrary != null && (String.Compare(runtimeLibrary.EvaluatedValue, "MultiThreadedDebugDLL", true) == 0 ||
+                                                String.Compare(runtimeLibrary.EvaluatedValue, "MultiThreadedDebug", true) == 0);
+        }
+
+        TaskOutput = useDebug;
+        
+        ]]>
```

Comment:
pretty impressive bit of config magic! And good attention to detail.

Reply:
you should see how cswinrt does string processing 😊